gustaf/io/meshio.py

```diff
@@ -72,18 +72,65 @@ def load(fname):
     return mesh
 
 
-def export(mesh, fname):
-    """Currently not implemented function.
+def export(mesh, fname, submeshes=None):
+    """Export mesh elements and vertex data into meshio and use its write
+    function. The definition of submeshes with identical vertex coordinates
+    is possible. In that case vertex numbering and data from the main mesh
+    are used.
 
     Parameters
     ------------
-    mesh: MESH_TYPES
-      Mesh to be exported.
+    mesh: Edges, Faces or Volumes
+      Input mesh
     fname: Union[str, pathlib.Path]
       File to save the mesh in.
+    submeshes: Iterable
+      Submeshes where the vertices are identical to the main mesh. The element
+      type can be identical to mesh.elements or lower-dimensional (e.g.
+      boundary elements).
 
     Raises
     -------
-    NotImplementedError: This method is currently not implemented.
+    NotImplementedError:
+       For mesh types that are not implemented.
+    ValueError:
+       Raises a value error, if the vertices indexed in a subset are not
+       present in the main mesh.
     """
-    raise NotImplementedError
+
+    # Mapping between meshio cell types and gustaf cell types
+    meshio_dict = {
+        "edges": "line",
+        "tri": "triangle",
+        "quad": "quad",
+        "tet": "tetra",
+        "hexa": "hexahedron",
+    }
+
+    cells = []
+    # Merge main mesh and submeshes in one list
+    meshes = [mesh]
+    if submeshes is not None:
+        meshes.extend(submeshes)
+
+    # Iterate the meshes and extract the element information in meshio format
+    for m in meshes:
+        whatami = m.whatami
+        if whatami not in meshio_dict.keys():
+            raise NotImplementedError(
+                f"Sorry, we can't export {whatami}-shape with meshio."
+            )
+        else:
+            if np.any(m.elements.flatten() > len(m.vertices)):
```

Comment:
```suggestion
            if np.any(m.elements > len(m.vertices) - 1):
```
I think you don't need to flatten the array. Also, the maximum reference value should be one less than the len(vertices), since ids start from 0